### PR TITLE
Add Scatter graph

### LINKED ISSUE #215: Implement Scatter chart

### PR DESCRIPTION
**Pull request by Alteoris**
Add Scatter graph.
Allow to hide the curve for scatter graph.
Can specify the tick label position.
Allow to invert graph order. 

Close #215

### CODE REVIEW

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs [hunk 1]
```diff
@@ -54,5 +54,15 @@ public class ChartAxisModel
...
+
+        /// <summary>
+        /// Define Axis orientation and minimum and maximum values
+        /// </summary>
+        public ScalingModel ScalingModel { get; set; } = new ScalingModel();
+
+        /// <summary>
+        /// Define the tick label position
+        /// </summary>
+        public TickLabelPositions? TickLabelPositions { get; set; }
```
Comment (HisCodeness) [inline OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs:66]: Are you sure this is plural ?

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/TickLabelPositions.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿namespace OpenXMLSDK.Engine.Word.ReportEngine.Models.Charts
+{
+    /// <summary>
+    /// Define the position of tick labels
+    /// </summary>
+    public enum TickLabelPositions
```
Comment (HisCodeness) [inline OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/TickLabelPositions.cs:6]: It's better to use singular

**Review by HisCodeness — APPROVED** [review]

**Review by DamienSAVOYAT — APPROVED** [review]
I add some comments optional before completing Pull Request.

File: OpenXMLSDK.Engine/Word/ReportEngine/Renders/LineChartExtensions.cs [hunk 3]
```diff
@@ -180,7 +181,7 @@ public static void ManageLineChart(LineModel chartModel, PlotArea plotArea, UInt
                     new AxisPosition() { Val = new EnumValue<AxisPositionValues>(AxisPositionValues.Bottom) },
                     new MajorTickMark() { Val = TickMarkValues.None },
                     new MinorTickMark() { Val = TickMarkValues.None },
-                    new TickLabelPosition() { Val = new EnumValue<TickLabelPositionValues>(TickLabelPositionValues.Low) },
+                    new TickLabelPosition() { Val = chartModel.CategoriesAxisModel.TickLabelPosition.HasValue ? new EnumValue<DC.TickLabelPositionValues>((DC.TickLabelPositionValues)(int)chartModel.CategoriesAxisModel.TickLabelPosition) : new EnumValue<DC.TickLabelPositionValues>(DC.TickLabelPositionValues.NextTo) },
```
[the remark this answers is not in the document]
Comment (Alteoris) [reply OpenXMLSDK.Engine/Word/ReportEngine/Renders/LineChartExtensions.cs:184]: Both enum don't coming from the same Namespace, I need to transform it to int to cast it in the DC. enum

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartModel.cs [hunk 4]
```diff
@@ -134,17 +134,34 @@ public bool ShowDataLabel
         /// <summary>
         /// Define the model for categories axis
```
[the remark this answers is not in the document]
Comment (Alteoris) [reply OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartModel.cs:135]: I'm not really sure that will be useful. It's the classic axes position, category on bot, values on right ;)

**Pull request merged**